tests/ui/indexing_slicing.stderr

```diff
@@ -91,181 +83,123 @@ error: range is out of bounds
-
-error: slicing may panic.
-  --> $DIR/indexing_slicing.rs:43:6
-   |
-43 |     &empty[1..5];
-   |      ^^^^^^^^^^^
-   |
-   = help: Consider using `.get(n..m)` or `.get_mut(n..m)` instead
-
 error: range is out of bounds
   --> $DIR/indexing_slicing.rs:45:6
    |
-45 |     &empty[..=4];
+45 |     &empty[1..5];
    |      ^^^^^^^^^^^
 
-error: slicing may panic.
-  --> $DIR/indexing_slicing.rs:45:6
-   |
-45 |     &empty[..=4];
-   |      ^^^^^^^^^^^
-   |
-   = help: Consider using `.get(..n)`or `.get_mut(..n)` instead
-
 error: range is out of bounds
-  --> $DIR/indexing_slicing.rs:50:6
-   |
-50 |     &empty[..=0];
-   |      ^^^^^^^^^^^
-
-error: slicing may panic.
-  --> $DIR/indexing_slicing.rs:50:6
+  --> $DIR/indexing_slicing.rs:47:6
    |
-50 |     &empty[..=0];
+47 |     &empty[..=4];
    |      ^^^^^^^^^^^
-   |
-   = help: Consider using `.get(..n)`or `.get_mut(..n)` instead
 
 error: range is out of bounds
   --> $DIR/indexing_slicing.rs:52:6
    |
-52 |     &empty[1..];
-   |      ^^^^^^^^^^
-
-error: slicing may panic.
-  --> $DIR/indexing_slicing.rs:52:6
-   |
-52 |     &empty[1..];
-   |      ^^^^^^^^^^
-   |
-   = help: Consider using `.get(n..)` or .get_mut(n..)` instead
+52 |     &empty[..=0];
+   |      ^^^^^^^^^^^
 
 error: range is out of bounds
-  --> $DIR/indexing_slicing.rs:53:6
+  --> $DIR/indexing_slicing.rs:54:6
    |
-53 |     &empty[..4];
+54 |     &empty[1..];
    |      ^^^^^^^^^^
 
-error: slicing may panic.
-  --> $DIR/indexing_slicing.rs:53:6
+error: range is out of bounds
+  --> $DIR/indexing_slicing.rs:55:6
    |
-53 |     &empty[..4];
+55 |     &empty[..4];
    |      ^^^^^^^^^^
-   |
-   = help: Consider using `.get(..n)`or `.get_mut(..n)` instead
 
 error: indexing may panic.
-  --> $DIR/indexing_slicing.rs:56:5
+  --> $DIR/indexing_slicing.rs:58:5
    |
-56 |     v[0];
+58 |     v[0];
    |     ^^^^
    |
    = help: Consider using `.get(n)` or `.get_mut(n)` instead
 
 error: indexing may panic.
-  --> $DIR/indexing_slicing.rs:57:5
+  --> $DIR/indexing_slicing.rs:59:5
    |
-57 |     v[10];
+59 |     v[10];
    |     ^^^^^
    |
    = help: Consider using `.get(n)` or `.get_mut(n)` instead
 
 error: slicing may panic.
-  --> $DIR/indexing_slicing.rs:58:6
+  --> $DIR/indexing_slicing.rs:60:6
    |
-58 |     &v[10..100];
+60 |     &v[10..100];
    |      ^^^^^^^^^^
    |
    = help: Consider using `.get(n..m)` or `.get_mut(n..m)` instead
 
 error: slicing may panic.
-  --> $DIR/indexing_slicing.rs:59:6
+  --> $DIR/indexing_slicing.rs:61:6
    |
-59 |     &v[10..];
+61 |     &v[10..];
    |      ^^^^^^^
    |
    = help: Consider using `.get(n..)` or .get_mut(n..)` instead
 
 error: slicing may panic.
-  --> $DIR/indexing_slicing.rs:60:6
+  --> $DIR/indexing_slicing.rs:62:6
    |
-60 |     &v[..100];
+62 |     &v[..100];
    |      ^^^^^^^^
    |
    = help: Consider using `.get(..n)`or `.get_mut(..n)` instead
 
-error: aborting due to 36 previous errors
+error: aborting due to 28 previous errors
```

Comment:
The errors should have gone to 37... Looks like something got bumped in the `return` refactoring. I'm investigating now.

Reply:
37? That would be 1 more error message. I think 28 could be right. Before the last commit multiple lines triggered both the `INDEXING_SLICING` and the `OUT_OF_BOUNDS_INDEXING` Lint. Now there should only be the `OUT_OF_BOUNDS_INDEXING` Lint in this cases.

But I didn't really look at the .stderr file yet.

Reply:
:thinking:

Ah totally. You're right about the number going down. 

I'm not certain it's still catching everything it should though. Minimally I'm going to push a change that makes that test file a little clearer about what should and shouldn't produce stderr so it's not so easy to get lost.